src/Aspire.Hosting.Azure/Extensions/AzureSqlExtensions.cs

```diff
@@ -71,4 +72,107 @@ private static IResourceBuilder<AzureSqlServerResource> ConfigureDefaults(this I
         return builder.WithManifestPublishingCallback(resource.WriteToManifest)
                       .WithParameter("serverName", resource.CreateBicepResourceName());
     }
+
+    internal static IResourceBuilder<SqlServerServerResource> PublishAsAzureSqlDatabaseConstruct(this IResourceBuilder<SqlServerServerResource> builder, Action<ResourceModuleConstruct, SqlServer, IEnumerable<SqlDatabase>>? configureResource = null, bool useProvisioner = false)
+    {
+        var configureConstruct = (ResourceModuleConstruct construct) =>
+        {
+            var sqlServer = new SqlServer(construct, builder.Resource.Name);
+
+            sqlServer.AssignProperty(x => x.Administrators.AdministratorType, "'ActiveDirectory'");
+            sqlServer.AssignProperty(x => x.Administrators.IsAzureADOnlyAuthenticationEnabled, "true");
+            sqlServer.AssignParameter(x => x.Administrators.Sid, construct.PrincipalIdParameter);
+            sqlServer.AssignParameter(x => x.Administrators.Login, construct.PrincipalName);
+            sqlServer.AssignProperty(x => x.Administrators.TenantId, "subscription().tenantId");
+
+            // TODO: This specially named firewall rule allows access to this database from all other
+            //       Azure services. There is a problem here. The constructor of SqlFirewall does not
+            //       take a parent object so it infers which SQL instance we are using from the
+            //       construct. if we have two sql instances in the construct that would be a problem
```

Comment:
ack - will add a parent param

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/42341/commits/44f5067f77547eda27632eb39a2a9b5f664733a6